pkg/build/controller/factory/factory.go

```diff
@@ -261,6 +267,49 @@ func (factory *ImageChangeControllerFactory) Create() controller.RunnableControl
 	}
 }
 
+type BuildConfigControllerFactory struct {
+	Client                  osclient.Interface
+	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+	// Stop may be set to allow controllers created by this factory to be terminated.
+	Stop <-chan struct{}
+}
+
+// buildConfigControllerRetry is the retry function for BuildConfigController
+// If the number of retries exceeds maxRetries it returns false to stop retrying
+func buildConfigControllerRetry(obj interface{}, err error, retries controller.Retry) bool {
+	bc := obj.(*buildapi.BuildConfig)
+	if retries.Count > maxRetries {
+		glog.V(3).Infof("Giving up retrying BuildConfig update event %s/%s: %v", bc.Namespace, bc.Name, err)
+		kutil.HandleError(err)
+		return false
+	}
+	glog.V(4).Infof("Retrying BuildConfig update event %s/%s: %v", bc.Namespace, bc.Name, err)
+	return true
+}
```

Comment:
`buildPodControllerRetry` is identical to this method, since the only reason you need to cast to particular object (Pod or BuildConfig) is to get its Namespace and Name, I'd suggest creating one generic method and get the Namespace/Name with [cache.MetaNamespaceKeyFunc](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/store.go#L75) from k8s.